src/seq.rs

```diff
@@ -0,0 +1,102 @@
+/// Conference, Pages 292-298, https://doi.org/10.1145/3652892.3700766
+#[derive(Clone, Debug, Default)]
+struct SeqChunkerState {
+    /// The length of the current incrementing run.
+    increment_run_length: usize,
+
+    /// The previously ingested byte, if any.
+    previous_value: Option<u8>,
+
+    /// Current slope length.
+    slope_length: usize,
+
+    /// The position _relative to the last chunk boundary_.
+    /// Specifically, this resets to zero after a chunk is emitted.
+    /// We need this to track if we're done filling our window between calls to `find_boundary`,
+    /// since we're fed blocks of data which may be smaller than the window.
+    pos: usize,
+}
+
+impl SeqChunkerState {
+    fn reset(&mut self) {
+        self.increment_run_length = 1;
+        self.previous_value = None;
+        self.slope_length = 1;
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SeqChunker {
+    /// The length of the current incrementing run.
+    seq_length: usize,
+
+    /// The number of consecutive increasing bytes required to trigger a chunk boundary.
+    skip_trigger: usize,
+
+    /// The number of bytes to skip ahead when the skip trigger is met.
+    skip_size: usize,
+
+    /// The current state of the chunker.
+    state: SeqChunkerState,
+}
+
+impl SeqChunker {
+    pub fn new(seq_length: usize, skip_trigger: usize, skip_size: usize) -> Self {
+        SeqChunker {
+            seq_length,
+            skip_trigger,
+            skip_size,
+            state: Default::default(),
+        }
+    }
+}
+
+impl ChunkerImpl for SeqChunker {
+    /// Searches for a chunk boundary in the provided data block.
+    /// If a boundary is found that is not the very end of the data block,
+    /// its index is returned within an Option.
+    fn find_boundary(&mut self, data: &[u8]) -> Option<usize> {
+        let mut i = self.state.pos;
+        while i < data.len() {
+            let b = data[i];
+            if let Some(previous_value) = self.state.previous_value {
+                if b > previous_value {
+                    // We have an incrementing interval.
+                    self.state.increment_run_length += 1;
+                    if self.state.increment_run_length == self.seq_length {
+                        return Some(i);
+                    }
+                } else if self.state.slope_length == self.skip_trigger {
+                    // Skip ahead by advancing the index
+                    i += self.skip_size;
+                    self.state.reset();
+                    continue;
+                } else {
+                    self.state.increment_run_length = 1;
+                    self.state.slope_length += 1;
+                }
+            }
+            self.state.previous_value = Some(b);
+            i += 1;
+        }
+
+        // No cut-point found within the current data block.
+        // Advance offset.
+        self.state.pos += data.len();
+        None
+    }
+
+    fn reset(&mut self) {
```

Comment:
This is called after a chunk is emitted, to reset the state of the chunker (see [here](https://github.com/mrd0ll4r/cdchunking-rs/blob/master/src/lib.rs#L159-L160)). That means it probably also needs to reset `self.state.pos`. This would also be an elegant place to implement skipping the beginning of each block :)